### PR TITLE
Made BaseRepo methods async

### PR DESCRIPTION
**Pull request by sirojbekdev**

### CODE REVIEW

File: src/BudgetManager.Shared/BudgetManager.Shared.DataAccess.MongoDB/BaseImplementation/BaseRepository.cs [hunk 1]
```diff
@@ -19,9 +19,9 @@ public BaseRepository(IMongoDbSettings settings, IMongoClient client)
             _collection = database.GetCollection<TDocument>(GetCollectionName(typeof(TDocument)));
         }
 
-        public virtual Task<List<TDocument>> GetAllAsync(CancellationToken cancellationToken)
+        public virtual async Task<IEnumerable<TDocument>> GetAllAsync(CancellationToken cancellationToken)
         {
-            return Task.Run(() => _collection.Find(_ => true).ToListAsync(cancellationToken));
+            return await _collection.Find(_ => true).ToListAsync(cancellationToken);
```
Comment (VladislavGromadtsov) [inline src/BudgetManager.Shared/BudgetManager.Shared.DataAccess.MongoDB/BaseImplementation/BaseRepository.cs:24]: we need to make every Find -> FindAsync

**Review by VladislavGromadtsov — CHANGES_REQUESTED** [review]
Also please add DTOs folder in SDK project

File: src/BudgetManager.Shared/BudgetManager.Shared.DataAccess.MongoDB/BaseImplementation/BaseRepository.cs (hunk 1 above)
Comment (VladislavGromadtsov) [reply src/BudgetManager.Shared/BudgetManager.Shared.DataAccess.MongoDB/BaseImplementation/BaseRepository.cs:24]: Can we make it?

Comment (otash8110) [reply src/BudgetManager.Shared/BudgetManager.Shared.DataAccess.MongoDB/BaseImplementation/BaseRepository.cs:24]: No, it is not possible, ToListAsync is only way to make it asynchronous

**Review by VladislavGromadtsov — APPROVED** [review]

**Review by otash8110 — APPROVED** [review]

**Review by Saigunas — APPROVED** [review]

**Pull request merged**